### DISCUSSION TITLE
Support for extruded tilesets

### DISCUSSION THREAD

**Discussion by psawaya** [replit/kaboom#641] [Ideas]
(Building off of an earlier conversation in #4) 

Is there any plan for Kaboom to support extruded tilesets?

By that, I mean tilesets that have a little bit of margin between each tile. This prevents "tile bleed". For an example, see a screenshot from my project below: 
<img width="313" alt="Screen Shot 2022-10-07 at 7 00 05 PM" src="https://user-images.githubusercontent.com/172380/194682419-160d5eb3-d97a-4162-8594-e8cc3f7d81e4.png">

See also this project from the Phaser community that takes an ordinary tile set and extrudes it: https://www.npmjs.com/package/tile-extruder

I think implementing it in Kaboom would be really easy. I'm currently drawing my tilemap by loading the entire map in as a sprite, and rendering a particular tile as a `frame` with `drawSprite`. The likely change would be adding optional `margin` and `spacing` parameters to `drawSprite`.

If there's interest for this, I might be able to take a swing at implementing this myself. :)

**Comment by lajbel** [comment]
Maybe. For now you can add a little extra size inside the `addLevel`

For example, every tile in this example are 64x64, and in `width` and `height` I put 65, this add a margin of 1 pixel 

![image](https://user-images.githubusercontent.com/71136486/194707220-96f88d6a-602d-4268-ae42-262f878c739f.png)
